### PR TITLE
User can access their profile page

### PR DESCRIPTION
**Pull request by mantanquzzmo**
[PT-link](https://www.pivotaltracker.com/story/show/170673756)

- Adds Profile page using Axios

Skärmavbild 2020-01-24 kl  12 39 44

### CODE REVIEW

File: src/modules/getUserData.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import axios from "axios";
+
+const getUserData = async id => {
+  let headers = JSON.parse(localStorage.getItem("J-tockAuth-Storage"));
+  let response = await axios
+    .get(`/admin/users/${id}`,
+    {
+      headers: headers
+    })
+    .then(response => {
```
Comment (oliverochman) [inline src/modules/getUserData.js]: why are you not using a `try, catch` block here instead?
keep it consistent

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: cypress/integration/userProfilePage.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    cy.get("#loginButton").click();
+    cy.get("#login").within(() => {
+      cy.get("#email").type("user@mail.com");
+      cy.get("#password").type("password");
+      cy.get("button")
+        .contains("Submit")
+        .click();
+    });
+    cy.get("#login").within(() => {
+      cy.get("a")
```
Comment (oliverochman) [inline cypress/integration/userProfilePage.js]: what is "a"?

File: src/components/DisplayProfile.jsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+  useEffect(() => {
+    userDataGrab();
+  }, [props.userAttrs]);
+
+  return (
+    <div id="profile">
+      {props.userShowData ? (
+        props.userShowData.email ? (
+          <>
```
Comment (oliverochman) [inline src/components/DisplayProfile.jsx:21]: what is the difference between the first two conditions here?

File: src/components/Signup.jsx [hunk 4]
```diff
@@ -16,7 +16,7 @@ const Signup = props => {
         props.changeAuthMessage(`Welcome! ${userDatas.data.data.email}`);
       })
       .catch(error => {
-        if (error.message == "Request failed with status code 422") {
+        if (error.message === "Request failed with status code 422") {
```
Comment (oliverochman) [inline src/components/Signup.jsx]: I must have missed this in the previous PR, why are you not using the real error response from the backend?

Comment (oliverochman) [reply src/components/Signup.jsx]: I also see that you are using a `then, catch` here.  We try to move away from old syntax as much as possible. Especially when we use async methods so much in this project.

File: src/components/Signup.jsx [hunk 5]
```diff
@@ -54,6 +54,7 @@ const Signup = props => {
         </>
       );
       break;
+    default: signupFunction = null
```
Comment (oliverochman) [inline src/components/Signup.jsx:53]: what is this?

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: cypress/integration/userProfilePage.js (hunk 2 above)
Comment (mantanquzzmo) [reply cypress/integration/userProfilePage.js]: anchor

File: src/components/Signup.jsx (hunk 5 above)
Comment (mantanquzzmo) [reply src/components/Signup.jsx:53]: Working through the error messages coming from inspecting in the browser.  It said it expects a default case for switch statements

File: src/components/DisplayProfile.jsx (hunk 3 above)
Comment (mantanquzzmo) [reply src/components/DisplayProfile.jsx:21]: First makes sure that userShowData exists, if not then it renders please log in. Second one checks if it has properly been loaded with the showData and not only the error message.

File: src/modules/getUserData.js (hunk 1 above)
Comment (mantanquzzmo) [reply src/modules/getUserData.js]: fixed

File: src/components/Signup.jsx (hunk 4 above)
Comment (mantanquzzmo) [reply src/components/Signup.jsx]: this is the syntax that comes with j-tockauth

Comment (mantanquzzmo) [reply src/components/Signup.jsx]: Isnt that the error message coming from axios/jtockauth?

Comment (oliverochman) [reply src/components/Signup.jsx]: https://github.com/CraftAcademy/ca-cooper-backend/blob/updated_dec_2019/spec/requests/api/v1/registrations_spec.rb

here you have the error messages that devise sends back

File: src/components/Signup.jsx (hunk 5 above)
Comment (oliverochman) [reply src/components/Signup.jsx:53]: okay cool

File: cypress/integration/userProfilePage.js (hunk 2 above)
Comment (oliverochman) [reply cypress/integration/userProfilePage.js]: is there a possibility you can put an id on this element?

File: src/components/DisplayProfile.jsx (hunk 3 above)
Comment (oliverochman) [reply src/components/DisplayProfile.jsx:21]: okay cool

File: cypress/integration/userProfilePage.js (hunk 2 above)
Comment (mantanquzzmo) [reply cypress/integration/userProfilePage.js]: there is. added.

File: src/components/Signup.jsx (hunk 4 above)
Comment (mantanquzzmo) [reply src/components/Signup.jsx]: sorted

**Pull request merged**